tests/harness/mod.rs

```diff
@@ -0,0 +1,211 @@
+
+    let alice2 = alice1.receive(message1_bob).unwrap();
+    let bob2 = bob1.receive(message1_alice).unwrap();
+
+    let message2_alice = alice2.next_message();
+    let message2_bob = bob2.next_message();
+
+    let alice3 = alice2.receive(message2_bob).unwrap();
+    let bob3 = bob2.receive(message2_alice).unwrap();
+
+    let message3_alice = alice3.next_message();
+    let message3_bob = bob3.next_message();
+
+    let alice4 = alice3.receive(message3_bob).unwrap();
+    let bob4 = bob3.receive(message3_alice).unwrap();
+
+    let message4_alice = alice4.next_message();
+    let message4_bob = bob4.next_message();
+
+    let alice5 = alice4.receive(message4_bob).unwrap();
+    let bob5 = bob4.receive(message4_alice).unwrap();
+
+    let message5_alice = alice5.next_message(alice_wallet).await.unwrap();
+    let message5_bob = bob5.next_message(bob_wallet).await.unwrap();
+
+    let alice6 = alice5.receive(message5_bob, alice_wallet).await.unwrap();
+    let bob6 = bob5.receive(message5_alice, bob_wallet).await.unwrap();
+
+    Created {
+        alice: alice6,
+        bob: bob6,
+    }
+}
+
+pub struct UpdateActors {
+    pub alice: update::Party0,
+    pub bob: update::Party0,
+}
+
+pub fn make_update_actors(alice: create::Party6, bob: create::Party6) -> UpdateActors {
+    let alice = update::Party0::new(alice);
+    let bob = update::Party0::new(bob);
+
+    UpdateActors { alice, bob }
+}
+
+pub struct Updated {
+    pub alice: update::Party0,
+    pub bob: update::Party0,
+}
+
+pub fn run_update_protocol(
+    alice0: update::Party0,
+    bob0: update::Party0,
+    channel_update: ChannelUpdate,
+    time_lock: u32,
+) -> Updated {
+    let (alice1, message0) = alice0
+        .propose_channel_update(channel_update, time_lock)
+        .unwrap();
+
+    let (bob1, message1) = bob0.receive_channel_update(message0).unwrap();
+
+    let alice2 = alice1.receive(message1).unwrap();
+
+    let message2_alice = alice2.next_message();
+    let message2_bob = bob1.next_message();
+
+    let alice3 = alice2.receive(message2_bob).unwrap();
+    let bob2 = bob1.receive(message2_alice).unwrap();
+
+    let message3_alice = alice3.next_message();
+    let message3_bob = bob2.next_message();
+
+    let alice4 = alice3.receive(message3_bob).unwrap();
+    let bob3 = bob2.receive(message3_alice).unwrap();
+
+    let message4_alice = alice4.next_message();
+    let message4_bob = bob3.next_message();
+
+    let alice5 = alice4.receive(message4_bob).unwrap();
+    let bob4 = bob3.receive(message4_alice).unwrap();
+
+    Updated {
+        alice: alice5,
+        bob: bob4,
+    }
+}
+
+pub struct Wallet(pub bitcoin_harness::Wallet);
+
+impl Wallet {
+    pub async fn new(name: &str, url: Url) -> anyhow::Result<Self> {
+        let wallet = bitcoin_harness::Wallet::new(name, url).await?;
+
+        Ok(Self(wallet))
+    }
+}
+
+#[async_trait::async_trait]
+impl BuildFundingPSBT for Wallet {
+    async fn build_funding_psbt(
```

Comment:
Would it make sense to move that to `bitcoin-harness-rs`?

Reply:
I don't think so. The trait is part of the public API of the create protocol. I don't see why `bitcoin-harness-rs` would want to implement a trait for a specific protocol.

Reply:
Ah sorry, did not realise it was a trait impl!